File: est datos/python/insertion_sort.py
def insercion(arr):
    for i in range(1, len(arr)):
        temp = arr[i]
        j = i-1

        while j >= 0 and temp < arr[j]:
            arr[j+1] = arr[j]
            j= j-1
        arr[j+1] = temp

def printarr(arr):
    for i in range(len(arr)):
        print(arr[i], end = " ")

arr = [70,15,2,51,60,32]
print("Arreglo sin ordenar")
printarr(arr)
insercion(arr)
print("\nArreglo ordenado")
printarr(arr)

File: est datos/c#/insertion_sort.cs
using System;
using System.ComponentModel.Design;
class Program{
    static void insercion(ref int[] arr)
    {
        for (int i = 0; i < 5; i++)
        {
            int temp = arr[i];
            int j = i - 1;

            while (j >= 0 && temp < arr[j])
            {
                arr[j + 1] = arr[j];
                j = j - 1;
            }
            arr[j + 1] = temp;
        }
    }

    static void imprimir(int[] arr)
    {
        for (int i = 0; i < 5; i++)
        {
            Console.WriteLine(arr[i] + string.Join(","));
        }
    }

    static void Main(string[] args)
    {
            int[] arr = {34,76,3,23,6};
            Console.WriteLine("Arreglo sin ordenar: \n");
            imprimir(arr);
            insercion(ref arr);
            Console.WriteLine("Arreglo ordenado: \n");
            imprimir(arr);
 
    }
}